src/smartstream/examples/map_regex/src/lib.rs

```diff
@@ -0,0 +1,20 @@
+use fluvio_smartstream::{smartstream, Record, RecordData};
+use regex::Regex;
+
+#[smartstream(map)]
+pub fn map(record: &Record) -> (Option<RecordData>, RecordData) {
+    let key = record.key.clone();
+
+    let string_result = std::str::from_utf8(record.value.as_ref());
```

Comment:
So maybe we should have conversion to string like this: `let string_result: String  = record.value.try_into()?`.
This also indicates that we need to return Result: `Result<(Option<RecordData>, RecordData)>` where Result will be smart stream result type.

Reply:
If we made the return type `-> Result<(Option<RecordData>, RecordData)>`, what would you expect the value in the stream to be when an `Err` is returned?

Example:

```
Topic: integers =          [16, 9, 4, 1, 0, -1]
Consumer squareroot.wasm = [4, 3, 2, 1, 0, ?]
```

If we had a consumer stream that took the square root of all the input numbers, then we can imagine our `squareroot` function returns `Err` when given a number less than 0. How would we want that to be represented in the output stream?

---

In the `filter`-style SmartStreams, the way I handled this was by early-returning `false` whenever some step failed, essentially filtering out all malformatted data. For Map I sort of ended up following the same approach, where if any step fails we simply return the input data unmodified. It does seem like we might want to have a better way of handling this.

Reply:
Error handling strategy tracked in #1198